cli/src/cli_util.rs

```diff
@@ -3847,6 +3848,18 @@ fn map_clap_cli_error(mut cmd_err: CommandError, ui: &Ui, config: &StackedConfig
     let Some(err) = cmd_err.error.downcast_ref::<clap::Error>() else {
         return cmd_err;
     };
+    if let Some(ContextValue::String(cmd)) = err.get(ContextKind::InvalidSubcommand) {
+        let cmd = cmd.as_str();
+        if matches!(cmd, "clone" | "init") {
+            // git commands
+            return cmd_err.clone().hinted(formatdoc!(
```

Comment:
nit: `.clone()` shouldn't be needed. You can `cmd_err.add_hint(..)` and return.

Reply:
Sure. I hope the resulting `cmd.clone()` doesn't look too confusing, but it is cheaper than `cmd_err.clone()`.